samlauthenticator/samlauthenticator.py

```diff
@@ -559,6 +578,17 @@ def _get_username_from_saml_etree(self, signed_xml):
         self.log.warning('Could not find name from name XPath')
         return None
 
+    def _get_roles_from_saml_etree(self, signed_xml):
+        xpath_with_namespaces = self._make_xpath_builder()
+        xpath_fun = xpath_with_namespaces(self.xpath_role_location)
+        xpath_result = xpath_fun(signed_xml)
+
+        if xpath_result:
+            return xpath_result
+
+        self.log.warning('Could not find role from role XPath')
+        return None
```

Comment:
Instead of returning `None` here, you could also return `[]` (empty list). That would keep the semantic of (new) line 603 (the if user_roles, return user_roles) since empty list is still False-y in Python, and then you wouldn't have to have the default value on (new) line 658 (since the default would be the thing that you want to pass back). If you're convinced that `None` is the right choice here, we can keep it.

Reply:
This is a better way of doing it. I have updated to return an empty array over None.